### DISCUSSION TITLE
Is Ray still alive?

### DISCUSSION THREAD

**Discussion by arne-s** [spatie/ray#941] [General]
I noticed there were only two bugfixes this year, no new features or plans. Is there still active development? It's hard to justify recurring payments if there are no updates.

**Comment by patinthehat** [comment]
@arne-s The next major version of the ray application is currently under development,  so I expect there will be more activity on the repo once it's released. 

Are there any features you feel are missing currently from this package?

**Reply by arne-s** [reply]
@patinthehat Great to hear there's a new version coming. I use Ray daily but it started to feel like it was abandoned. There are many features I would like to see, I'm not sure if this is the right place, but here are a few:

- Speed improvement, Ray often becomes unresponsive and glitchy when there are many logs. 
- Allow logs to be shown collapsed/compact, this is especially true for exceptions, they take up so much space because of the tracelog, that it drowns out all other messages. I would like logs to be displayed in a very compact way, and when you click it, you can see the full entry, this should also speed up the rendering. 
- SSH connections should be restricted to one user. I have a server with multiple websites, each has its own user. But when i connect to one account, I also get the logs from other sites on the same server. So either it should only show logs from the current user, or you should be able to filter on the user you want to see. 
_edit: now that I think about it, is the current situation not a huge security vulnerability when used on shared hosting? I often leave ray() calls in the code, couldn't another user on the same server see that data?_
- Tag support: basically the same as specifying a color, but instead of a color you specify one or more tags, and you can filter on those tags in the app.  e.g. `ray($var)->tags('form', 'validation')`. Also allow a default tag to be specified in the ray.php so all logs on that installation have a certain tag. (e.g. `DEFAULT_TAGS = ['production', 'projectname']`;). In the future you could even create an option to attach a behavior to a tag, like displaying logs expanded for a certain tag, or play a sound for a different tag. 
- This one is a nice to have, but it would be useful if we could log an image path, and render it as an image in Ray. Something like `ray()->image('/tmp/hdbwjfb.png');` The same thing could be done for downloading a file: `ray()->file('/tmp/order_99434.docx');` and ray shows it as a download link in the app. 
- Allow logs to be exported to a text file.

**Reply by patinthehat** [reply]
ping @freekmurze @sebastiandedeyne

**Reply by lcallif** [reply]
Yepper, sure looks like this is now abandonware...  No notifications of a new release, nothing buttttt an email stating when the license is up and that it would be billed at/after that date.  Danggggggg  

Given all of their fantastic set of products (mostly shareware) I guess I shouldn't complain.  But, jezzzzzzzz   :-(

**Reply by patinthehat** [reply]
@lcallif Spatie is working on the next major version as far as I know. Stay tuned!

**Comment by freekmurze** [comment]
Hi, my apologies for the lack of updates.

We are currently rebuilding Ray from the ground up, and we will soon entering our public beta phase. 
We'll post [updates on the Ray blog](https://myray.app/blog/sign-of-life).
